lib/tasks/keel.rake

```diff
@@ -1,246 +1,246 @@
-
-			unless pods.length > 0
-				message = "Could not find pods in the \"#{deploy_env}\" environment."
-				prompter.print message, :error
-				abort
-			end
-
-			# It seems that the first pod is the one we really want to look at for logs
-			pod = pods.first
-			pod.logs tail
-		end
+namespace :keel do
+  def pick_sha(sha)
+    unless @sha
+      prompter  = Keel::GCloud::Prompter.new
+      @sha = prompter.prompt_for_sha sha
+    end
+    @sha
+  end
+
+  def pick_namespace(namespace)
+    unless @namespace
+      prompter  = Keel::GCloud::Prompter.new
+      # Fetch namespaces from k8s
+      namespaces 	= Keel::GCloud::Kubernetes::Namespace.fetch_all
+      unless namespaces
+        message = 'Unable to connect to Kubernetes, please try again later...'
+        prompter.print message, :error
+        abort
+      end
+      @namespace  = prompter.prompt_for_namespace namespaces, namespace
+    end
+    @namespace
+  end
+
+  desc "build a docker image suitable for pushing"
+  task :pack, [:deploy_sha] do |_, args|
+    config    = Keel::GCloud::Config.new
+
+    #TODO warn about the madness of dirty folders, and the fact that we aren't really checking anything!
+    pack_sha  = pick_sha(args[:deploy_sha])
+    command   = "docker build -t gcr.io/#{config.project_id}/#{config.app_name}:#{pack_sha} ."
+    puts command
+    Keel::GCloud::Cli.new.execute(command)
+  end
+
+  desc "ship the image to gcloud"
+  task :push, [:deploy_sha] do |_, args|
+    config    = Keel::GCloud::Config.new
+
+    #TODO figure out if the sha is something we can actually deploy
+    pack_sha  = pick_sha(args[:deploy_sha])
+    command   = "gcloud docker push gcr.io/#{config.project_id}/#{config.app_name}:#{pack_sha}"
+    puts command
+    Keel::GCloud::Cli.new.execute(command)
+  end
+
+  desc "provision a deployment and service on kubernetes"
+  task :provision, [:deploy_sha] do |_, args|
+    config    	= Keel::GCloud::Config.new
+    prompter  	= Keel::GCloud::Prompter.new
+    pack_sha  	= pick_sha(args[:deploy_sha])
+    
+    deploy_env  = pick_namespace args[:environment]
+
+    # Retrieve a replication controller configuration from the cluster
+    rcs = Keel::GCloud::Kubernetes::ReplicationController.fetch_all deploy_env, config.app_name
+    if rcs
+      message = "Found an existing deployment or replication controller for #{config.app_name}"
+      prompter.print message, :error
+    else
+      message = Keel::GCloud::Kubernetes::ReplicationController.create namespace, config.app_name, config.project_id, "3000", pack_sha
+      prompter.print message, :info
+    end
+  end
+
+    task :shipit, [:deploy_sha] => [:set_gcloud_properties,
+                                  :pack,
+                                  :push,
+                                  :provision,
+                                  :deploy] do |_, args|
+    puts "packed, pushed, provisioned and deployed #{pick_sha(args[:deploy_sha])}!"
```

Comment:
For consistency (and my OCD), this would be better if it were printed out as a success (like the other ones, in green).
The prompter handles these things for us.